### PR TITLE
bump version to 0.17.1

### PR DESCRIPTION
**Pull request by mrocklin**

### CODE REVIEW

**Comment by jakirkham** [comment]
So we should take this opportunity to make one change. Namely [switch to using `pip`]( https://conda-forge.org/docs/meta.html#use-pip ). This is particularly important here as this is something we are trying to do with `noarch: python` packages.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,17 +7,18 @@ package:
...
 build:
   number: 0
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  script: python -m pip install --no-deps --ignore-installed .
   noarch: python
 
 requirements:
   build:
     - python
     - setuptools
```
Comment (jakirkham) [inline recipe/meta.yaml]: Minor point: We can drop `setuptools` now as `pip` handles that. Though not a big deal either way.

Comment (mrocklin) [reply recipe/meta.yaml]: Easy enough to fix.  Thanks for shepherding me through this :)

**Comment by mrocklin** [comment]
Replicating the pip changes over at https://github.com/conda-forge/distributed-feedstock/pull/43

**Comment by jakirkham** [comment]
Thanks. LGTM.

**Pull request merged**